src/com/perl5/lang/perl/idea/run/PerlRunProfileState.java

```diff
@@ -155,4 +163,54 @@ protected ProcessHandler startProcess() throws ExecutionException
 		return runProfile.getEnvs();
 	}
 
+	public List<String> getPerlLibraries(@Nullable VirtualFile scriptFile)
+	{
+		if (!PlatformUtils.isIntelliJ())
+		{
+			return Collections.emptyList();
+		}
+		else
+		{
+			PerlConfiguration runProfile = (PerlConfiguration) getEnvironment().getRunProfile();
+
+			// try to get used SDK
+			Sdk sdk;
+			if (runProfile.isUseAlternativeSdk())
+			{
+				sdk = ProjectJdkTable.getInstance().findJdk(runProfile.getAlternativeSdkPath());
+			}
+			else
+			{
+				Module moduleForFile = scriptFile == null ? null : ModuleUtilCore.findModuleForFile(scriptFile, getEnvironment().getProject());
+
+				sdk = PerlRunUtil.getModuleSdk(moduleForFile);
+				if (sdk == null)
+				{
+					// try project SDK
+					sdk = ProjectRootManager.getInstance(getEnvironment().getProject()).getProjectSdk();
+				}
+			}
+
+			if (sdk != null && sdk.getSdkType() == PerlSdkType.getInstance())
+			{
+				List<String> incPaths = PerlSdkType.getInstance().getINCPaths(sdk.getHomePath());
+				List<String> list = new ArrayList<>();
+
+				for (VirtualFile file : sdk.getRootProvider().getFiles(OrderRootType.CLASSES))
```

Comment:
Yes I am using it to receive the classroots from SDK that is already "module-specific" so it handles the module or project setting. It returns all entries that are listed in the classpath section in the SDK settings including the built-in one. (This one is removed with the help of `PerlSdkType.getInstance().getINCPaths(sdk.getHomePath());`)

The `com.intellij.openapi.roots.libraries.LibraryUtil#getLibraryRoots(...)` only returns the built-in one and can only be used for the for modules and project SDKs and not for the alternative one.

Of course `PerlSharedSettings#getLibRootUrls` must be excluded too.